File: Data Structures & Algorithms/house-robber/submission-1.java
class Solution {
    public int rob(int[] nums) {
        if(nums.length == 1)
        {
            return nums[0];
        }
        int[] values = new int[nums.length];
        for(int i =  0; i < values.length; i++)
        {
            values[i] = -1;
        }
        dfs(0, nums.length, nums, values);
        dfs(1, nums.length, nums, values);
        return Math.max(values[0], values[1]);
    }
    public int dfs(int i, int length, int[] nums, int[] values)
    {
        if(i >= length)
        {
            return 0;
        }
        if(values[i] != -1)
        {
            return values[i];
        }
        values[i] = Math.max(dfs(i + 2, length, nums,values), dfs(i + 3, length, nums, values)) + nums[i];
        return values[i];
    }
}

File: Data Structures & Algorithms/kth-largest-element-in-an-array/submission-1.py
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        i,length = 0,len(nums)
        while i < len(nums):
            val = max(nums)
            k = k - 1
            if k == 0:
                return val
            j = nums.index(val)
            del nums[j]
        